src/hooks/useLtRepoValidation.js

```diff
@@ -0,0 +1,115 @@
+import { useEffect, useState } from 'react';
+import YAML from 'js-yaml-parser'
+import {
+  HTTP_GET_MAX_WAIT_TIME,
+} from '@common/constants'
+import {
+  doFetch,
+  isServerDisconnected,
+  onNetworkActionButton,
+  processNetworkError,
+  reloadApp,
+} from '@utils/network'
+import {decodeBase64ToUtf8} from '@utils/decode'
+
+
+
+export default function useLtRepoValidation({authentication, owner, server, languageId}) {
+
+  const [ltRepoTree, setLtRepoTree] = useState({})

```

Comment:
Cecil great job with this hook, I just have one recommendation. I suggest combining the three useState into one useState using an object to avoid unnecessary extra re-renders that may subsequently affect app performance.